### ISSUE TITLE
Build time up to 15x slower when REMOVING the @nestjs/swagger plugin from nest-cli.json

### ISSUE THREAD

**Issue by christian-gama** [nestjs/nest-cli#2833]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

I have a monorepo project that uses webpack + swc-loader + @nestjs/swagger plugin. Everything runs fine and fast (build is around 100ms), however if I simply REMOVE the @nestjs/swagger plugin, build time increases to ~=2200ms for no reason. 

<img width="1587" alt="Screenshot 2024-12-04 at 09 01 42" src="https://github.com/user-attachments/assets/10148cda-95a4-4858-946c-21869c7db81d">

<img width="964" alt="Screenshot 2024-12-04 at 09 15 06" src="https://github.com/user-attachments/assets/fb124763-4f77-4156-bb1c-f89545ebdd3d">

### Minimum reproduction code

https://github.com/christian-gama/nestjs-build-bug-reproduction

### Local
1. Run `git clone https://github.com/christian-gama/nestjs-build-bug-reproduction.git`
2. Run `cd nestjs-build-bug-reproduction.git`
3. Run `pnpm install`
4. Run `chmod +x ./entrypoint.sh`
5. Run `./entrypoint.sh`
6. Compare the build time with plugin enabled and plugin disabled

### Docker
1. Run `git clone https://github.com/christian-gama/nestjs-build-bug-reproduction.git`
2. Run `cd nestjs-build-bug-reproduction.git`
3. Run `pnpm docker:build`
4. Compare the build time with plugin enabled and plugin disabled

### Expected behavior

Disabling or enabling the @nestjs/swagger should have no impact in the build time of the application, but the weirdest behavior is to have a build time 10x faster with the plugin ENABLED, which does not make any sense, since swc-loader does not even support the @nest/swagger plugin.

I'd like to see a faster build time without using the @nest/swagger plugin enabled - the only thing this plugin is doing for the project is to improve the build time, which shouldn't be happening.

### Package

- [ ] I don't know. Or some 3rd-party package
- [ ] <code>@nestjs/common</code>
- [ ] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [ ] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [X] Other (see below)

### Other package

@nestjs/swagger

### NestJS version

10.4.12

### Packages versions

[System Information]
OS Version     : macOS 24.1.0
NodeJS Version : v23.1.0
NPM Version    : 10.9.0

[Nest CLI]
Nest CLI Version : 10.4.8

[Nest Platform Information]
platform-express version : 10.4.12
cache-manager version    : 2.3.0
microservices version    : 10.4.12
mapped-types version     : 2.0.6
schematics version       : 10.2.3
passport version         : 10.0.3
schedule version         : 4.1.1
terminus version         : 10.2.3
swagger version          : 8.0.7
testing version          : 10.4.12
common version           : 10.4.12
config version           : 3.3.0
axios version            : 3.1.2
core version             : 10.4.12
cli version              : 10.4.8
 version                 : 2.4.2
 version                 : 1.2.4

### Node.js version

23.1.0

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by kamilmysliwiec** [comment]
I'll look into it as soon as I can, but it has low priority as—as I've already mentioned above—plugin and swc-loader are not compatible anyway, so the results you're seeing are incorrect (there's very likely some error being thrown under the hood that falls back to a different configuration—a plugin entry in the configuration might be causing it, but I'll have to double-check).

**Comment by kamilmysliwiec** [comment]
Alright, I had a few minutes and decided to dive deeper into this issue.

The difference is, when a project doesn't use any plugin, we can run the `fork-ts-checker-webpack-plugin` (https://github.com/TypeStrong/fork-ts-checker-webpack-plugin) package to verify TS types in parallel (while TS/SWC itself just transpiles your code without performing any type checking). However, if you have at least 1 plugin registered, then we can't use that package and we need to rely on TSC (or one of the NestJS CLI plugins themselves).

That being said, in your repository `npm run build-with-plugin` command **does not perform any type-checking** (it just transpiles your code). For example, if you change `app.listen` to `app.random` it won't even fail (your build will succeed) - that's why it's "faster". On the other hand, if you run `npm run build-without-plugin` you'll see errors:

![image](https://github.com/user-attachments/assets/c0b57bf1-8984-4d55-aaee-316a76536ff1)

Hence 3 seconds instead of 400ms

**Comment by kamilmysliwiec** [comment]
Why would you ever want `npm run build` to run without a type-checker? For --watch mode you won't see much of a difference anyway

**Comment by christian-gama** [comment]
> Why would you ever want `npm run build` to run without a type-checker? For --watch mode you won't see much of a difference anyway

I've been trying to create a NestJS project using Bun and sometimes I got a different behavior using `bun --bun nest start` than running the application directly from build, so I was developing relying on `bun nest build --watch` and `bun --watch dist/main.js`, which always gave me the correct and expected behavior. Having a long build time and having to wait a few seconds for every minor change that I made is painful, thats why I'd like to have typecheck disabled.

**Comment by kamilmysliwiec** [comment]
```js
const swcDefaultConfig =
  require('@nestjs/cli/lib/compiler/defaults/swc-defaults').swcDefaultsFactory()
    .swcOptions;

module.exports = (webpackConfig) => {
  return {
    module: {
      rules: [
        {
          test: /\.ts$/,
          exclude: /node_modules/,
          use: {
            loader: 'swc-loader',
            options: swcDefaultConfig,
          },
        },
      ],
    },
    plugins: [webpackConfig.plugins[0]],
  };
};
```

![image](https://github.com/user-attachments/assets/184239b8-c4cb-4f79-ad97-1aa5d965aa2f)

☝️ this configuration excludes `ForkTsCheckerPlugin`
